python-sdk/src/astro/table.py

```diff
@@ -26,13 +26,24 @@ class Metadata:
     """
 
     # This property is used by several databases, including: Postgres, Snowflake and BigQuery ("namespace")
-    schema: str | None = None
+    _schema: str | None = None
     database: str | None = None
+    location: str | None = None
 
     def is_empty(self) -> bool:
         """Check if all the fields are None."""
         return all(getattr(self, field_name) is None for field_name in fields_dict(self.__class__))
 
+    @property
+    def schema(self):
+        if self.location:
+            return f"{self._schema}__{self.location.replace('-', '_')}"
```

Comment:
How would this look when `self._schema` is empty?

Reply:
Why are we modifying `self.location`? Would using `-` break something?

Reply:
Added a comment explaining the need for replacement.